### ISSUE TITLE
[Button] Cannot focus button in Safari with keyboard navigation

### ISSUE THREAD

**Issue by nikki-lea** [Esri/calcite-design-system#7972]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

By pressing tab, I should be able to toggle focused elements on the page to focus a calcite button.

### Expected Behavior

In Safari, tabbing elements never results in focusing any calcite buttons on the page.

### Reproduction Sample

The button on calcite-react-template reproduces this

### Reproduction Steps

1. Open [codesandbox with the calcite-react-template](https://codesandbox.io/s/ecstatic-pine-44yz6x) in the Safari browser
2. Click the white background of the page with the React Template title
3. Tab through elements on the react page
4. Note that only the slider receives focus

### Reproduction Version

1.9.1

### Relevant Info

MacOS, Safari 16

### Priority impact

p3 - want for upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Enterprise

**Comment by jcfranco** [comment]
Worth noting that Safari, unfortunately, doesn't tab through some elements by default (see [toy repro case](https://codepen.io/jcfranco/pen/rNoPXjv)) and requires enabling a specific OS setting.

Did a quick search and there are some existing issues related to this:

* https://bugs.webkit.org/show_bug.cgi?id=199671
* https://bugs.webkit.org/show_bug.cgi?id=18425

**Comment by jcfranco** [comment]
> and requires enabling a specific OS setting.

Confirmed this and the behavior you expect depends on users enabling the `Keyboard navigation` option under the `Keyboard` preference pane:

![Screenshot 2023-11-27 at 3 03 17 PM](https://github.com/Esri/calcite-design-system/assets/197440/1df7f4af-f235-4d62-af85-fa0cb9d02a75)

https://support.apple.com/sq-al/guide/mac-help/mchlp1399/mac#:~:text=With%20keyboard%20navigation%2C%20you%20can,may%20need%20to%20scroll%20down).&text=Turn%20on%20%E2%80%9CKeyboard%20navigation%E2%80%9D%20on%20the%20right.

Closing as this is working as expected once the above option is enabled.
